spark-common/src/main/java/com/k15t/spark/atlassian/AtlassianIframeAppServlet.java

```diff
@@ -0,0 +1,124 @@
+        // load contentWindow part of the iFrameResizer (inject as inline script), otherwise left the app untouched
+        String iframeResizerContentWindowJs = DocumentOutputUtil.getIframeResizeContentWindowJs();
+        document.head().append("\n<script>\n" + iframeResizerContentWindowJs + "\n</script>\n");
+    }
+
+
+    /**
+     * <p>
+     * Parse (or generate) a document to be used as the wrapper for the SPA with an iframe and needed decorators
+     * to get the admin view.
+     * </p><p>
+     * Adds the Atlassian admin decorator meta element and a content decorator element telling which webitem should
+     * be marked as selected. Then adds as the main body an iframe element that will ask loading the actual SPA.
+     * </p><p>
+     * Also adds iFrameResizer main window part to the page and some inline JavaScript needed for communicating all
+     * the context values to the contents of the iframe.
+     * </p>
+     *
+     * @param document {@link Document} of the index before processing
+     * @param props {@link RequestProperties} of the current request
+     */
+    private void prepareAdminIframeWrapperIndex(Document document, RequestProperties props) throws IOException {
+        // add atlassian admin decorator meta element
+        Element metaElement = document.head().appendElement("meta");
+        metaElement.attr("name", "decorator");
+        metaElement.attr("content", "atl.admin");
+
+        // remove all the styles and wrappers from the iframe parent, they will only be needed in the actual content iframe
+        document.head().children().not("title,meta,content").remove();
+
+        // fill the body with an iframe that will ask the actual app with 'iframe_content' query parameter
+        // remove all other content - it will be shown when the app is loaded with iframe_content parameter
+
+        document.body().children().remove();
+
+        Element contEl = document.body().appendElement("content");
+        contEl.attr("tag", "selectedWebItem");
+        contEl.text(getSelectedWebItemKey());
+
+        String iframeHtml = getTemplateRenderer().renderFragment(
+                DocumentOutputUtil.getIframeAdminContentWrapperTemplate(),
+                DocumentOutputUtil.generateAdminIframeTemplateContext(
+                        props.getRequest().getRequestURI(), "spark_admin_iframe",
+                        "admin", null, props.getRequest().getQueryString()));
+
+        document.body().append(iframeHtml);
+    }
+
+
+    /**
+     * Return the key of the web item. Needed for selecting the correct item in the admin side menu. The default implementation
+     * returns the value of servlet init parameter {@link Keys#SPARK_SELECTED_WEB_ITEM_KEY}
+     *
+     * @return key of the web item that should be marked as selected in the admin side menu
+     */
+    public String getSelectedWebItemKey() {
+        return StringUtils.defaultString(
```

Comment:
This should first check for a request query parameter of name ```Keys.SPARK_SELECTED_WEB_ITEM_KEY``` and read the servlet init param only if no such query parameter exists.

This way we could add the key of the highlighted web-item to the ```<link>``` as specified *within* the web-item and thereby reuse the same servlet for multiple web items in the global admin area.